### PR TITLE
feat: Introduce admin users

### PR DESCRIPTION
**Pull request by olerichter00**
Addresses [CX-1647]

### Description

- Adds admin users to replace the `ADMIN` and `CATALOGUERS` consts.
- Adds admin users interface for CRUD actions.

Screenshot 2021-07-12 at 19 27 40

### Follow Ups

- Run `bundle exec rake admin_users:create`
- Remove `config/initializers/global_constants.rb`

[CX-1647]: https://artsyproduct.atlassian.net/browse/CX-1647

### CODE REVIEW

File: app/views/admin/admin_users/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<div class='container'>
+
```
Comment (olerichter00) [inline app/views/admin/admin_users/_form.html.erb:2]:
```suggestion
```

**Comment by Gon4ar5** [comment]
https://github.com/artsy/convection/blob/olerichter00/CX-1647/add-admin-model/app/views/admin/submissions/show.html.erb#L115

I think it needs to be changed from
`<% options = [['Please select one']] + CATALOGUERS.collect { |grav_id, name| [name, grav_id] } %>`
to
`<% options = [['Please select one']] + AdminUser.cataloguers.map { |cataloguer| [cataloguer.name, cataloguer.gravity_user_id] } %>`

**Comment by Gon4ar5** [comment]
> https://github.com/artsy/convection/blob/olerichter00/CX-1647/add-admin-model/app/views/admin/submissions/show.html.erb#L115
> 
> I think it needs to be changed from
> `<% options = [['Please select one']] + CATALOGUERS.collect { |grav_id, name| [name, grav_id] } %>`
> to
> `<% options = [['Please select one']] + AdminUser.cataloguers.map { |cataloguer| [cataloguer.name, cataloguer.gravity_user_id] } %>`

Is it possible to leave comments on unchanged code on github? I do not understand😅

**Comment by olerichter00** [comment]
Good catch, @Gon4ar5 ! I forgot to update the cataloguers. I updated all appearances  of `CATALOGUERS`

**Comment by Gon4ar5** [comment]
And also (purely from my visual point of view😁 ) it would be better to make the buttons of the same size and "align: center" on some elements, something like this
Screenshot 2021-07-15 at 17 09 44

**Review by Gon4ar5 — APPROVED** [review]
Very cool! Thank you for what you did🔥 🔥 🔥

**Comment by olerichter00** [comment]
> And also (purely from my visual point of view😁 ) it would be better to make the buttons of the same size and "align: center" on some elements, something like this
> Screenshot 2021-07-15 at 17 09 44

@Gon4ar5 Yes, that looks way better. Can you push these changes? I just kept it as simple and didn't pay too much attention to how things look...

File: app/views/admin/admin_users/index.html.erb [hunk 2]
```diff
@@ -11,41 +11,41 @@
...
       <%= admin_user.cataloguer ? "Yes" : "No" %>
     </div>
-    <div class='col-md-1'>
+    <div class='col-md-1 align-center'>
       <%= admin_user.super_admin ? "Yes" : "No" %>
     </div>
-    <div class='col-md-1'>
-      <%= button_to 'Edit', edit_admin_admin_user_path(admin_user), method: :get, class: 'btn btn-small btn-primary pull-right' %>
+    <div class='col-md-2'%>
+      <%= button_to '&emsp;Edit&emsp;'.html_safe, edit_admin_admin_user_path(admin_user), method: :get, class: 'btn btn-small btn-primary pull-right', style: 'float: center;'%>
```
Comment (Gon4ar5) [inline app/views/admin/admin_users/index.html.erb:45]: Not really sure if this approach is correct, but I haven't come up with anything better🙈

**Review by Gon4ar5 — APPROVED** [review]

File: app/views/admin/admin_users/index.html.erb (hunk 2 above)
Comment (olerichter00) [reply app/views/admin/admin_users/index.html.erb:45]: 👍

File: lib/tasks/offers_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require "rails_helper"
```
Comment (joeyAghion) [inline lib/tasks/offers_spec.rb:3]: I think this is causing staging deploys [to fail recently](https://app.circleci.com/pipelines/github/artsy/convection?branch=master). It should go under `spec/` if it's a test, so the deployed environments don't try to load it.

**Pull request merged**